share/wake/lib/system/remote_cache_runner.wake

```diff
@@ -0,0 +1,357 @@
+
+                require Some (Pass stdout) =
+                    blobDownloads
+                    | mlookup stdout_blob.getRemoteCacheBlobId
+                else failWithError "unable to download stdout"
+
+                require Some (Pass stderr) =
+                    blobDownloads
+                    | mlookup stderr_blob.getRemoteCacheBlobId
+                else failWithError "unable to download stderr"
+
+                def resolved_outputs =
+                    output_files
+                    | map (_.getCacheSearchOutputFilePath)
+
+                def outputs = resolved_outputs ++ output_dirs ++ output_symlinks
+                def predict = Usage status runtime cputime mem ibytes obytes
+                def _ = virtual job stdout stderr status runtime cputime mem ibytes obytes
+                def inputs = map getPathName (input.getRunnerInputVisible)
+
+                Pass (RunnerOutput inputs outputs predict)
+
+            def _ = breadcrumb "Did not find a match"
+
+            # -------------------------------------
+            # --- Insert the job into the cache ---
+            # -------------------------------------
+
+            # Run the job to get the results
+            require Pass output = baseDoIt job (Pass input)
+
+            # Post the job to the server. This is left as 'def _' so that wake won't block progess
+            # on it but it will stil be joined on before wake exits. Regardless of the result this
+            # call should return the passing output to allow the build to continue.
+            def _ = postJob rscApi job wakeroot hashKey input output
+
+            Pass output
+
+    Runner "remote-cache: {name}" score doit
+
+## --- Helper functions ---
+
+# strToBytes "foo" = (13, 14, 15, Nil)
+def strToBytes (str: String): List Integer =
+    str
+    | explode
+    # FIXME: unicodeToInteger may return values larger than u8 which will break the serialization
+    | map (unicodeToInteger _)
+
+# listFlattenWithNull ((1,2,3,), (4,5,6,), (7,8,9,),) = (1,2,3,0,4,5,6,0,7,8,9,)
+def listFlattenWithNull (parts: List (List Integer)): List Integer =
+    foldr (\x \acc x ++ (0, acc)) Nil parts
+
+# asBytesDelimedByNull ("a", "b", "cd",) = (15, 0, 16, 0, 17, 18)
+def asBytesDelimedByNull (parts: List String): List Integer =
+    parts
+    | map strToBytes
+    | listFlattenWithNull
+
+# getPathAsJson (Path "foo" "asdf") = (JObject ...)
+def getPathAsJson (Path path hash) =
+    JObject (
+        "path" :-> JString path,
+        "hash" :-> JString hash,
+    )
+
+# toJIntegerList (1,2,3,) = (JInteger 1, JInteger 2, JInteger 3,)
+def toJIntegerList (parts: List Integer): List JValue =
+    parts
+    | map JInteger
+
+# Posts a completed job to the remote cache
+def postJob (rscApi: RemoteCacheApi) (job: Job) (_wakeroot: String) (hidden: String) ((RunnerInput _label cmd vis env dir stdin _res _prefix _ isAtty): RunnerInput) ((RunnerOutput _ outputs usage): RunnerOutput): Result Unit Error =
```

Comment:
From here below, there is a lot of JSON management. I'm going to move that into `remote_cache_api.wake` but I wanted to get this posted first so folks could start looking at it